standalone/test/integration_tests/api/integrationtest_scanner_api.cpp

```diff
@@ -181,14 +251,14 @@ TEST_F(ScannerAPITests, shouldReceiveStartRequestWithCorrectHostIpWhenUsingAutoI
   setUpScannerV2();
   setUpStrictScannerMock();
   const auto start_req_received_barrier =
-      prepareStrictMockStartRequestBarrier(generateScannerConfig(HOST_IP_ADDRESS, FRAGMENTED_SCAN));
+      prepareStartRequestBarrier(generateScannerConfig(HOST_IP_ADDRESS, FRAGMENTED_SCAN));
 
   strict_scanner_mock_->startListeningForControlMsg();
   const auto start_future = scanner_->start();
 
   ASSERT_TRUE(start_req_received_barrier->waitTillRelease(DEFAULT_TIMEOUT)) << "Start request not received";
```

Comment:
If you aim for readability I think lines like this are key. If I read this I have to think around a few corners. Actually I want to read that "Here the Mock is expected to receive a start request with some special config". Instead I read that some barrier is waiting to be release. Maybe this could be abstracted away.

Reply:
You are right, it would be nice if we could get that more clear. I would suggest to do that in a separate pr though